decidim-core/app/models/decidim/taxonomy_filter.rb

```diff
@@ -30,75 +40,95 @@ def self.log_presenter_class_for(_log)
 
     # Public name for this filter, defaults to the root taxonomy name.
     def name
-      return root_taxonomy.name if super&.compact_blank.blank?
+      return root_taxonomy&.name if super&.compact_blank.blank?
 
       super
     end
 
     # Internal name for this filter, defaults to the root taxonomy name.
     def internal_name
-      return root_taxonomy.name if super&.compact_blank.blank?
+      return root_taxonomy&.name if super&.compact_blank.blank?
 
       super
     end
 
+    def translated_name
+      decidim_sanitize_translated(name)
+    end
+
+    def translated_internal_name
+      decidim_sanitize_translated(internal_name)
+    end
```

Comment:
you could delegate it to the presenter, or you could refactor to something like 
```suggestion
    def translated_name
      presenter.name
    end

    def translated_internal_name
      presenter.internal_name
    end
```